Readme.md

```diff
@@ -138,36 +138,43 @@ addiu t3, t4, \
 OtherGlobalLabel:  ; this will terminate the area where
-                   ; @@LocalLabel can be used
-  b   @@LocalLabel ; as a result, this will cause an error
+                   ; @@LocalLabel can be used.
+  b   @@LocalLabel ; as a result, this will cause an error.
 ```
 
-Static labels behave like global labels, but are only valid in the very file they were defined. Any included files or files that include it cannot reference it. They can, however, contain another static label with the same name.
-
-```
-@StaticLabel:
-```
+Static labels behave like global labels, but are only valid in the very file they were defined. Any included files or files that include it cannot reference it. They can, however, contain another static label with the same name. When a static label is defined insde a (nested) macro, it is treated as being defined in the file where the top-level macro call occurred, rather than the file holding the macro definition where the static label is created.
 
 A label name can contain all characters from A-Z, numbers, and underscores. However, it cannot start with a digit. All label names are case insensitive.
 
-Additionally, `.` can be used to reference the current memory address.
+Additionally, `.` is a special label and can be used to reference the current memory address; it is equivalent to calling the expression function `org()`.
+
+A label can also be defined using the `.func`/`.function` directive. The example below will create a label `MyLabel` pointing to the current memory address. In addition, if the `-sym2` command line flag is used to output a sym2 file, the size of the function block (from `.func` to `.endfunc`) will also be written to the symfile.
+
+```
+.func MyLabel
+  ; assembly code
+.endfunc
+```
 
 ## 4.4 equ
 
-The `equ` directive works as a text replacement and is defined as follows:
+The `equ` directive works as a direct text replacement on the assembly source level and is defined as follows. Unlike labels, an `equ` must be defined before it can be used.
 
 ```
-@@StringPointer equ 0x20(r29)
+GlobalEqu   equ 1
+@StaticEqu  equ 2
+@@LocalEqu  equ 3
 ```
 
-There has to be a space before and after `equ`. The assembler will replace any occurrence of `@@StringPointer` with `0x20(r29)`. As it is a local `equ`, it will only do so in the current section, which is terminated by any global label or specific directives. This code:
+There has to be at least one whitespace character before and after `equ`. The assembler will replace any occurrence of `GlobalEqu`, `@StaticEqu` and `@@LocalEqu` with `1`, `2` and `3` respectively. Similarly to labels, a global `equ` is valid anywhere, a static `equ` is only valid in the file it was defined, and a local `equ` is only valid in the current section, which is terminated by any global or static label or specific directives. The replacement value can be an integer, a string, or even (partial) assembly code or directives. For example, this code:
```

Comment:
The replacement can be any sequence of valid tokens. Any usage of the name identifier is replaced by the respective tokens, in-place before any parsing is done.